### PR TITLE
Finish moving to intra doc links for std::sync

### PR DESCRIPTION
**Pull request by poliorcetics**
Helps with #75080.

@rustbot modify labels: T-doc A-intra-doc-links

r? @jyn514

### CODE REVIEW

File: library/std/src/sync/once.rs [hunk 1]
```diff
@@ -126,11 +124,8 @@ unsafe impl Sync for Once {}
 #[stable(feature = "rust1", since = "1.0.0")]
 unsafe impl Send for Once {}
 
-/// State yielded to [`call_once_force`]’s closure parameter. The state can be
-/// used to query the poison status of the [`Once`].
-///
-/// [`call_once_force`]: struct.Once.html#method.call_once_force
-/// [`Once`]: struct.Once.html
+/// State yielded to [`Once::call_once_force()`]’s closure parameter. The state
+/// can be used to query the poison status of the [`Once`].
```
[the remark this answers is not in the document]
Comment (poliorcetics) [reply library/std/src/sync/once.rs:128]: I'm not sure what to do about it. I thought up some alternate wordings but they are not any better IMO:

- `the Once instance`
- `the original Once`

[the remark this answers is not in the document]
Comment (jyn514) [reply library/std/src/sync/once.rs:128]: Yeah, I don't think it needs to change, it's just unfortunate. 'the Once' makes it clear you're talking about an instance of the struct, it just sounds weird.

File: library/std/src/sync/barrier.rs [hunk 2]
```diff
@@ -156,13 +150,12 @@ impl fmt::Debug for BarrierWaitResult {
 }
 
 impl BarrierWaitResult {
-    /// Returns `true` if this thread from [`wait`] is the "leader thread".
+    /// Returns `true` if this thread from [`Barrier::wait()`] is the
```
[the remark this answers is not in the document]
Comment (poliorcetics) [reply library/std/src/sync/barrier.rs]: I modified the wording to:

```
    /// Returns `true` if this thread is the "leader thread" for the call to
    /// [`Barrier::wait()`].
```

**Pull request merged**